swig/python/README.rst

```diff
@@ -4,38 +4,76 @@ GDAL/OGR in Python
 
 This Python package and extensions are a number of tools for programming and
 manipulating the GDAL_ Geospatial Data Abstraction Library.
-
-The GDAL project maintains SWIG generated Python
-bindings for GDAL/OGR. Generally speaking the classes and methods mostly
+Generally speaking the classes and methods mostly
 match those of the GDAL and OGR C++ classes. There is no Python specific
 reference documentation, but the https://gdal.org/api/python_bindings.html#tutorials includes Python examples.
 
+The GDAL Python package is built using `SWIG <https://www.swig.org>`__.
+The GDAL project maintains these Python bindings for GDAL/OGR within
+the gdal source tree.
+The SWIG-generated files are not checked in and are not part of the
+gdal release tarballs.  The bindings are also made available via PyPi
+where the swig-generated files are provided.
+
+Building from source
+--------------------
+
+To build from source, there are multiple possible paths.
+
 Dependencies
-------------
+~~~~~~~~~~~~
 
- * libgdal (3.9.0 or greater) and header files (gdal-devel)
- * numpy (1.0.0 or greater) and header files (numpy-devel) (not explicitly
-   required, but many examples and utilities will not work without it)
+(As usual, when a package is listed one needs the parts used to build
+programs against it, called -devel on many systems.  This is not
+special about gdal.)
 
-Installation
-------------
+ * SWIG (4 or greater).  
 
-Conda
-~~~~~
+ * libgdal (3.9.0 or greater).
 
-GDAL can be quite complex to build and install, particularly on Windows and MacOS.
-Pre built binaries are provided for the conda system:
+ * numpy (1.0.0 or greater).  This is not force-required by the build,
+   but many examples and utilities will not work without it.  At least
+   osgeo.gdal_array requires numpy at build time; building without
+   numpy and installing it later does not result in correct operation.
+   apparently not and in any case it is an unwarranted leap.
 
-https://docs.conda.io/en/latest/
+ * setuptools (python)
 
-By the conda-forge project:
+ * wheel (python)
 
-https://conda-forge.org/
+Building when building the rest of GDAL
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Once you have Anaconda or Miniconda installed, you should be able to install GDAL with:
+Python bindings are generated by default when building GDAL from source.
+For more detail, see `Python bindings options <https://gdal.org/development/building_from_source.html#building-python-bindings>`__
 
-``conda install -c conda-forge gdal``
+Building just the bindings, from the full tree
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+Another is to use the gdal source distribution with an
+already-installed gdal library and headers.  To do this, one must
+create a build directory and run cmake to generate makefiles.  Then,
+in swig/python, gmake python_generated_files.  Then, one can build
+using setup.py like any other distributed python module.
```

Comment:
rewritten in a way where this would work with other generators like Ninja:
```suggestion
already-installed gdal library and headers.  To do this, one must
create a build directory and run cmake to generate build scripts.  Then,
run ``cmake --build . --target python_generated_files``. Finally, in swig/python,
one can build using setup.py like any other distributed python module.
```